src/libs/OptionsListUtils.js

```diff
@@ -103,16 +103,10 @@ Onyx.connect({
  * @returns {Array}
  */
 function getPolicyExpenseReportOptions(report) {
-    if (!ReportUtils.isPolicyExpenseChat(report)) {
-        return [];
-    }
-    const filteredPolicyExpenseReports = _.filter(
-        policyExpenseReports,
-        (policyExpenseReport) => policyExpenseReport.policyID === report.policyID && policyExpenseReport.isOwnPolicyExpenseChat,
-    );
-    return _.map(filteredPolicyExpenseReports, (expenseReport) => {
-        const policyExpenseChatAvatarSource = ReportUtils.getWorkspaceAvatar(expenseReport);
-        return {
```

Comment:
> removing the early return

We only call `getPolicyExpenseReportOptions` if it's a policy expense chat, so it's kinda redundat.
<img width="1144" alt="image" src="https://github.com/Expensify/App/assets/50919443/dd629f6d-4dc0-4c75-8334-478bad45a8d2">

> and the filter here?

When we request money in a workspace chat, the participant is the workspace chat itself. We already have the `reportID`, so it's also kinda redundant to loop over all of our `policyExpenseReports` and find which report has the same `policyID` and it is `isOwnPolicyExpenseChat`. For example, I want to request money from this policy expense chat (isOwnPolicyExpenseChat is true for this chat):

<img width="375" alt="image" src="https://github.com/Expensify/App/assets/50919443/f636ce67-3361-42cf-a454-64f038081dd9">

Then, we loop over all policy expense chat and find which one has the same policyID, which includes this chat.
<img width="390" alt="image" src="https://github.com/Expensify/App/assets/50919443/d9fc33a8-981a-4ebe-8a28-5e8451fb34cd">

But the isOwnPolicyExpenseChat is false (we can't request money on above chat). At the end, we are doing a useless loop. That's why I also don't store the `policyID` in the Onyx because it is never used besides doing this filter.

Reply:
Indeed, it feels redundant, we were discussing this few weeks earlier. Can you confirm this does not recreate https://github.com/Expensify/App/issues/18708 ( check https://github.com/Expensify/App/pull/18215#issuecomment-1541909985 ).

I have a question though, isn't the passed `report` the same as `expenseReport`.

Reply:
Yep, I have checked that before and will never recreate that as we only construct a single element of array if it's a policy expense chat here:
<img width="941" alt="image" src="https://github.com/Expensify/App/assets/50919443/70e636f7-5a00-4a67-875e-380992588690">

> isn't the passed report the same as expenseReport

Yes, the `this.props.report` is the same as `expenseReport`

Reply:
Can we resolve this one?

Reply:
One question, since we do return `login` in `getParticipantsOptions`. Shouldn't we return `reportID` and `isPolicyExpenseChat` in `getPolicyExpenseReportOptions`?

Reply:
Ah, you're right. We should return the `isPolicyExpenseChat`. `reportID` will be available from `...expenseReport`. Will update this one!

Reply:
Updated!